sale_report_delivered_subtotal/reports/sale_report.py

```diff
@@ -8,17 +8,27 @@ class SaleReport(models.Model):
     _inherit = "sale.report"
 
     price_subtotal_delivered = fields.Float(string="Subtotal Delivered")
+    weight_delivered = fields.Float("Gross Weight Delivered")
 
     def _query(self, with_clause="", fields=None, groupby="", from_clause=""):
         if fields is None:
             fields = {}
-        select_str = """ ,
+        select_str_price = """ ,
             sum((l.price_subtotal /
                  coalesce(nullif(l.product_uom_qty, 0), 1)
                 ) * l.qty_delivered)
             as price_subtotal_delivered
         """
-        fields.update({"price_subtotal_delivered": select_str})
+        select_str_weight = """ ,
+            sum(p.weight * l.qty_delivered /
+                u.factor * u2.factor) as weight_delivered
```

Comment:
This is not quantity, so you shouldn't use UoM factors.

Reply:
Right it's already converted: https://github.com/odoo/odoo/blob/13.0/addons/sale/report/sale_report.py#L69

Reply:
I get this formula from this reference https://github.com/OCA/OCB/blob/25d0db569e26403777918ff179fe6d9993a694d5/addons/sale/report/sale_report.py#L95 changing the `product_uom_qty` by the `qty_delivered`